jobs/release-microk8s/release-microk8s.groovy

```diff
@@ -0,0 +1,131 @@
+                                } else if (env.ARCH == "amd64") {
+                                    instance_type = "m5.large"
+                                    constraints = "instance-type=${instance_type} arch=$ARCH root-disk=80G mem=16G cores=8"
+                                } else {
+                                    error("Aborting build due to unknown arch=${env.ARCH}")
+                                }
+                                sh """
+                                if ! timeout 4m juju destroy-controller -y --destroy-all-models --destroy-storage "${juju_controller}"; then
+                                   timeout 4m juju kill-controller -y "${juju_controller}" || true
+                                fi
+                                """
+                                sh """
+                                juju bootstrap "${env.JUJU_CLOUD}" "${juju_controller}" \
+                                    -d "${juju_model}" \
+                                    --model-default test-mode=true \
+                                    --model-default resource-tags=owner=k8sci \
+                                    --bootstrap-constraints "instance-type=${instance_type}"
+
+                                juju deploy -m "${juju_full_model}" --constraints "${constraints}" ubuntu
+
+                                juju-wait -e "${juju_full_model}" -w
+
+                                juju ssh -m "${juju_full_model}" --pty=true ubuntu/0 -- 'sudo snap install lxd'
+                                juju ssh -m "${juju_full_model}" --pty=true ubuntu/0 -- 'sudo lxd.migrate -yes' || true
+                                juju ssh -m "${juju_full_model}" --pty=true ubuntu/0 -- 'sudo lxd init --auto'
+                                """
+                                if (track == "pre-release"){
+                                    sh """
+                                    juju ssh -m "${juju_full_model}" --pty=true ubuntu/0 -- 'sudo snap install snapcraft --classic'
+                                    """
+                                }
+                                sh """
+                                . .tox/py38/bin/activate
+                                DRY_RUN=${params.DRY_RUN} ALWAYS_RELEASE=${params.ALWAYS_RELEASE} \
+                                    TRACKS=${params.TRACKS} TESTS_BRANCH=${params.TESTS_BRANCH} \
+                                    PROXY=${params.PROXY} JUJU_UNIT=ubuntu/0 \
+                                    timeout 6h python jobs/microk8s/release-to-${track}.py
+                                """
+                                sh "juju destroy-controller -y --destroy-all-models --destroy-storage ${juju_controller} || true"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    post {
+        always {
+            script {
+                all_tracks.each { track -> 
```

Comment:
this post is here in case any previous controllers weren't successfully cleaned up